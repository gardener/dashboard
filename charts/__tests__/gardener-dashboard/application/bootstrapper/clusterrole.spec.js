//
// SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and Gardener contributors
//
// SPDX-License-Identifier: Apache-2.0
//

'use strict'

const { helm } = fixtures

const renderTemplates = helm.renderBootstrapperApplicationTemplates

describe('gardener-dashboard-terminal-bootstrapper', () => {
  describe('clusterrole', () => {
    let values
    let templates

    beforeEach(() => {
      values = {
        global: {
          bootstrapper: {
            enabled: true
          }
        }
      }
      templates = [
        'clusterrole'
      ]
    })

    it('should not render the template', async () => {
      values.global.bootstrapper.enabled = false
      const documents = await renderTemplates(templates, values)
      expect(documents).toHaveLength(1)
      const [clusterRole] = documents
      expect(clusterRole).toBe(null)
    })

    it('should render the template', async () => {
      const documents = await renderTemplates(templates, values)
      expect(documents).toHaveLength(1)
      const [clusterRole] = documents
      expect(clusterRole).toMatchSnapshot()
    })
  })
})
