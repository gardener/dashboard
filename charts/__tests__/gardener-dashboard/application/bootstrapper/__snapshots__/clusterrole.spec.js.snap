// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gardener-dashboard-terminal-bootstrapper clusterrole should render the template 1`] = `
Object {
  "apiVersion": "rbac.authorization.k8s.io/v1",
  "kind": "ClusterRole",
  "metadata": Object {
    "labels": Object {
      "app.kubernetes.io/component": "terminal-bootstrapper",
      "app.kubernetes.io/instance": "gardener-dashboard",
      "app.kubernetes.io/managed-by": "Helm",
      "app.kubernetes.io/name": "gardener-dashboard",
      "helm.sh/chart": "gardener-dashboard-application-0.1.0",
    },
    "name": "gardener.cloud:system:dashboard-terminal-bootstrapper",
  },
  "rules": Array [
    Object {
      "apiGroups": Array [
        "",
      ],
      "resources": Array [
        "secrets",
      ],
      "verbs": Array [
        "get",
        "list",
      ],
    },
    Object {
      "apiGroups": Array [
        "core.gardener.cloud",
      ],
      "resources": Array [
        "shoots",
      ],
      "verbs": Array [
        "get",
        "list",
        "watch",
      ],
    },
    Object {
      "apiGroups": Array [
        "core.gardener.cloud",
      ],
      "resources": Array [
        "shoots/adminkubeconfig",
      ],
      "verbs": Array [
        "create",
      ],
    },
    Object {
      "apiGroups": Array [
        "seedmanagement.gardener.cloud",
      ],
      "resources": Array [
        "managedseeds",
      ],
      "verbs": Array [
        "get",
      ],
    },
  ],
}
`;
